From 6d19d559d29c1060c0d472f4d6a5daf2c5e608a9 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Mon, 7 Mar 2016 15:57:00 +0100
Subject: [PATCH] iproute: Descriptions of fou and gue options in ip-link man
 pages

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1013584
Upstream Status: iproute2.git commit 90f1df715eb38

commit 90f1df715eb38bbbb4554ebbad783d67690a38b2
Author: Tom Herbert <therbert@google.com>
Date:   Thu Jan 29 08:52:01 2015 -0800

    iproute: Descriptions of fou and gue options in ip-link man pages

    Add section for additional arguments to GRE, IPIP, and SIT types
    that are related to Foo-over-UDP and Generic UDP Encapsulation.
    Also, added an example GUE configuration in the examples section.

    Signed-off-by: Tom Herbert <therbert@google.com>
---
 man/man8/ip-link.8.in | 68 ++++++++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 67 insertions(+), 1 deletion(-)

diff --git a/man/man8/ip-link.8.in b/man/man8/ip-link.8.in
index 81ef4bb..9e667a2 100644
--- a/man/man8/ip-link.8.in
+++ b/man/man8/ip-link.8.in
@@ -460,6 +460,63 @@ Example:
 .in -8
 
 .TP
+GRE, IPIP, SIT Type Support
+For a link of types
+.I GRE/IPIP/SIT
+the following additional arguments are supported:
+
+.BI "ip link add " DEVICE
+.BR type " { gre | ipip | sit } "
+.BI " remote " ADDR " local " ADDR
+.R " [ "
+.BR encap " { fou | gue | none } "
+.R " ] [ "
+.BI "encap-sport { " PORT " | auto } "
+.R " ] [ "
+.BI "encap-dport " PORT
+.R " ] [ "
+.I " [no]encap-csum "
+.R " ] [ "
+.I " [no]encap-remcsum "
+.R " ]"
+
+.in +8
+.sp
+.BI  remote " ADDR "
+- specifies the remote address of the tunnel.
+
+.sp
+.BI  local " ADDR "
+- specifies the fixed local address for tunneled packets.
+It must be an address on another interface on this host.
+
+.sp
+.BR encap " { fou | gue | none } "
+- specifies type of secondary UDP encapsulation. "fou" indicates
+Foo-Over-UDP, "gue" indicates Generic UDP Encapsulation.
+
+.sp
+.BI "encap-sport { " PORT " | auto } "
+- specifies the source port in UDP encapsulation.
+.IR PORT
+indicates the port by number, "auto"
+indicates that the port number should be chosen automatically
+(the kernel picks a flow based on the flow hash of the
+encapsulated packet).
+
+.sp
+.I [no]encap-csum
+- specifies if UDP checksums are enabled in the secondary
+encapsulation.
+
+.sp
+.I [no]encap-remcsum
+- specifies if Remote Checksum Offload is enabled. This is only
+applicable for Generic UDP Encapsulation.
+
+.in -8
+
+.TP
 IP6GRE/IP6GRETAP Type Support
 For a link of type 
 .I IP6GRE/IP6GRETAP
@@ -495,7 +552,7 @@ the following additional arguments are supported:
 .sp
 .BI  local " ADDR "
 - specifies the fixed local IPv6 address for tunneled packets.
-It must be and address on another interface on this host.
+It must be an address on another interface on this host.
 
 .sp
 .BI  [i|o]seq
@@ -831,6 +888,15 @@ ip link help gre
 .RS 4
 Display help for the gre link type.
 .RE
+.PP
+ip link add name tun1 type ipip remote 192.168.1.1
+local 192.168.1.2 ttl 225 encap gue encap-sport auto
+encap-dport 5555 encap-csum encap-remcsum
+.RS 4
+Creates an IPIP that is encapsulated with Generic UDP Encapsulation,
+and the outer UDP checksum and remote checksum offload are enabled.
+
+.RE
 
 .SH SEE ALSO
 .br
-- 
1.8.3.1

